k8s-scheduler/src/main/java/org/dcm/PodEventsToDatabase.java

```diff
@@ -313,16 +319,53 @@ private void updatePod(final Pod pod) {
         return inserts;
     }
 
-    private boolean hasNodeSelector(final Pod pod) {
-        final PodSpec podSpec = pod.getSpec();
-        return  (podSpec.getNodeSelector() != null && podSpec.getNodeSelector().size() > 0)
-                || (podSpec.getAffinity() != null
-                && podSpec.getAffinity().getNodeAffinity() != null
-                && podSpec.getAffinity().getNodeAffinity()
-                .getRequiredDuringSchedulingIgnoredDuringExecution() != null
-                && podSpec.getAffinity().getNodeAffinity()
-                .getRequiredDuringSchedulingIgnoredDuringExecution()
-                .getNodeSelectorTerms().size() > 0);
+    private boolean hasNodeAffinityRequirements(final Pod pod, final boolean affinity) {
```

Comment:
Rather than have the return value encode true/false depending on "boolean affinity", use an enum for the parameter that allows you to write `if (searchingFor.equals(Affinity)) {....} else if (searchingFor.equals(AntiAffinity)) {...}`. It'll make add more clarity.